chainerio/io.py

```diff
@@ -99,38 +151,92 @@ def __exit__(self, exc_type: Optional[Type[BaseException]],
 
     @abstractmethod
     def isdir(self, file_path: str) -> bool:
+        """Returns ``Ture`` if the path is an existing directory
+
+        Args:
+            path (str): the path to the target directory
+
+        Returns:
+            ``True`` when the path points to a directory,
+            ``False`` when it is not
+
+        """
         raise NotImplementedError()
 
     @abstractmethod
     def mkdir(self, file_path: str, mode: int = 0o777,
               *args, dir_fd: Optional[int] = None) -> None:
+        """Makes a directory with mode
+
+        Args:
+            path (str): the path to the directory to make
+
+            mode (int): the mode of the new directory
+
+        """
         raise NotImplementedError()
 
     @abstractmethod
     def makedirs(self, file_path: str, mode: int = 0o777,
                  exist_ok: bool = False) -> None:
+        """Makes directories recursively with mode
+
+        Also creates all the missing parents of the given path.
+
+        Args:
+            path (str): the path to the directory to make.
+
+            mode (int): the mode of the directory
+
+            exist_ok (bool): In default case, a `FileExitsError` will be raised
+                when the target directory exists.
+                Set the ``exist_ok`` to surpass
+
+        """
         raise NotImplementedError()
 
     @abstractmethod
     def exists(self, file_path: str) -> bool:
+        """Returns ``True`` when the given ``path`` exists
+
+        Args:
+            path (str): the ``path`` to the target file. The ``path`` can be an
+            Unix path or an URI.
+
+        Returns:
+            ``True`` when the file or directory exists,
+            ``False`` when it is not.
+
+        """
         raise NotImplementedError()
 
     @abstractmethod
     def rename(self, src: str, dst: str) -> None:
+        """Renames the file from ``src`` to ``dst``
+
+        Args:
+            src (str): the current name of the target file or directory.
```

Comment:
The term "target" is confusing to me as in some context it can also be used for destination.